support/elasticsearch_search.erl

```diff
@@ -482,10 +482,11 @@ map_must({rsc_id, Id}, _Context) ->
     {true, #{<<"match">> => #{<<"_id">> => Id}}};
 map_must({text, "id:" ++ _ = Val}, Context) ->
     map_must({text, list_to_binary(Val)}, Context);
-map_must({text, <<"id:", Id/binary>>}, _Context) ->
-    %% Find by id when text argument equals "id:123"
+map_must({text, <<"id:", IdOrName/binary>>}, Context) ->
+    %% Find by id when text argument equals "id:123" or "id:unique_name"
+    Id = m_rsc:rid(z_string:trim(IdOrName), Context),
     {true, #{<<"match">> => #{
-        <<"_id">> => z_string:trim(Id)
+        <<"_id">> => Id
```

Comment:
And then, of course, we can just ditch the variable `Id`. 😉

Reply:
Hmm yes - somehow I liked it to have its own line, but indeed.